openpgp/ecdh_test.go

```diff
@@ -91,15 +87,121 @@ func openAndDecryptKey(t *testing.T, key string, passphrase string) EntityList {
+}
+
+const privKey521 = `-----BEGIN PGP PRIVATE KEY BLOCK-----
+
+lNkEWAJ/HhMFK4EEACMEIwQBX1achVr3ad6/1AYQM0Xpb0yOch0Va2+d1WjAi/TU
+lVMYFq3Sv1HRgwz87iaEGv2lViKTZ2Zbqh68ndyBoAY9CpQAzHrEnFozvQBQxSHe
+JaWxdiJIF3ZtLRrxMm+SBSKcQge2TwXmFr/coEKU3uS6PNHz9/1qKvOflbLwgiP6
+PWt01HYAAgUeo/x+60pfXvYBT/YwzYtEpMgY3ahEM64gNzCSwbggGdCK02H53Rir
+hQc4NHL/N/dYachvcGllNP2yi5ygNeSjYiDxtCRuaXN0IGtleSB0ZXN0ZXIgPG0r
+dGVzdGluZ0B6YXB1Lm5ldD6IvQQTEwoAIQUCWAJ/HgIbAwULCQgHAgYVCAkKCwIE
+FgIDAQIeAQIXgAAKCRDwOj0LJyvhfR5OAgkBhXIMxYkE8EuBDPjtHG7DliwBt+Ht
+++KWGHxWqkAFWQitjGK33JANOyuMjMr8ealisUsbRO4io51vsOa6BVrvQVsCCQEn
+VHpmetF7urR2j+V/Qr3SmT01sj0opToya52YoM1eS7+bSJRtPYyz4GomHSbMe76m
+zxqcXBu7xS1moh/HQP4gW5zeBFgCfx4SBSuBBAAjBCMEAP1NEe5jGggGOhGr99OX
+zwvBPLbcsyIf7cpqDi1IAHCxcnoYzVIoBJEjkdyHpuTQAvjddSF+SNGk48O4z+Ev
+tmlAAI1ChPg4ZLEk1fLqq/mxsyc3HT5Ny6cKYMeW3cfCAVLlmcLYPMt5ELCOBWj+
+Iy6fp22eVsMaL2S2teDJ+ZsN2abeAwEKCQACCQFm8eXql6OnFxTUQ1ODtW0ub4MM
+BNz1lcGW5PV06vXOwxKEcS1H3HK/ALqD3c7F+mQOAiWnmCXpNRgqKEfd1Rsz1SHH
+iKMEGBMKAAkFAlgCfx4CGwwACgkQ8Do9Cycr4X3dmAIHUn62iaxtsJ3/FlSZhXxy
+d8fW4Z3NhFlCLVL6p4NijQUJQPZMcDyh9fPvSdLE1CvBMtzow2qvEVUWiunus7nl
+mPwCBioXoB7rOhvEz59qnTLAjPLMOw9ib+IEjthSzrGJpfQVn1n/izJbfeG7Ghg+
+FAvmbYconl4Q0uWVJFs6Ys23JuUn
+=IypP
+-----END PGP PRIVATE KEY BLOCK-----`
+
+const privKey384 = `-----BEGIN PGP PRIVATE KEY BLOCK-----
+
+lKQEWAU34RMFK4EEACIDAwT2oNRSt6wQ2XR+yLsL5uLtmI+PGgaCpx84z1YSVRJg
+0/v1/7OnDpEmGv68GXajeZ5K4pDq31HNBrDEe5NwvLXCANOnSn4YQrYpCYVzJCgi
+yK67knfVPRTHuqfbrFrvi80AAYCjD4ZyVQ6aoSsWpD1KQYD9UFRrcpgwj8L3Rnmc
+1bt2KeRBRYLLWCLhzvWpKGrh/k4YS7QgTmlzdCAzODQgVGVzdGVyIDxtKzM4NEB6
+YXB1Lm5ldD6ImQQTEwkAIQUCWAU34QIbAwULCQgHAgYVCAkKCwIEFgIDAQIeAQIX
+gAAKCRC8jTWktJdxUg5CAX909W2xrO7CZa/M85D2yDY3r8vnjUaPN+aZbcYvYXeY
+QxsPdyvHQP+50OBz4KYW5coBgMA1TM7/nwL+y8vibwTuWlDtwQF2YY8MsBSeAl0Z
+6sZzAnwVTtHBJFZuCnHx1RxjF5yoBFgFN+ESBSuBBAAiAwMEwPrC34Behc3CLLfm
+z8nLWPM2RCNb+n+TweDf2vFYwSo96PQvecbB/KI7uthPtxVoecyLhRPR53FVp9Oc
+Ihfssx2uqVNsJD+0rG9RP7dFwhYjh2a45k0o0qpxXGXOPdusAwEJCQABgMA6AgN4
+XTfgtBJRRQ4GkTY1vFivGvwzq2qkansqEI6DLqqiOSqgtkhmgmnYFYOD5BRwiIEE
+GBMJAAkFAlgFN+ECGwwACgkQvI01pLSXcVJjyAF/TsbOlZHUQ/swWUUXHK+/ZGEv
+Sy+NVV6aThtefPRKdzFoOtVqm2zb0JiGoNTF0BbsAYCN3Z6tWZebr9Zv6I1H1w8U
+tWNM382gD7IUjkEz7BIWMbHkn4m8KsAxsax7VhmiTuo=
+=tJxL
+-----END PGP PRIVATE KEY BLOCK-----`
+
+const privKeyCv25519 = `-----BEGIN PGP PRIVATE KEY BLOCK-----
+
+lFgEV/bL8xYJKwYBBAHaRw8BAQdA/tN2DTMq9IDsDjE+d0jdrQv4nUh15IwhEuK6
+98RzHTAAAQC6gzVTi8V5Eis0pBg8g0iW0hp++dPczDXGg+Kc1jkzEw+ItB1NaWNo
+YWwgWiAoMjU1MTkpIDxtQHphcHUubmV0Poh5BBMWCAAhBQJX9svzAhsDBQsJCAcC
+BhUICQoLAgQWAgMBAh4BAheAAAoJEOctHO20d21/oE8A/jeDMoqnVrart8PlBBOh
+U7POysui1CFQb4bYokaURPNzAQCE7gJ0oD2pOlU6zgia1+6JPfAnUL8rQ4PFsZ7b
+5gT2BpxdBFf2y/MSCisGAQQBl1UBBQEBB0DR3in/BpS2e2jyZL1lX+DrUzJwXeQs
+6CTF+o83Jt32UgMBCAcAAP9tMK39aEcGzSUmICdAqybiurbh1anP453af1dwgiRb
+8BF8iGEEGBYIAAkFAlf2y/MCGwwACgkQ5y0c7bR3bX9JPgEA7WiFuFuTI4L0e8mV
+3UeahfoOyLOY71uHDNdfB66DCa0BANK4aMk+j7bpJoWFNpkWq9JnhpfXV9L2dh3R
+kKuKwZkI
+=RN+z
+-----END PGP PRIVATE KEY BLOCK-----`
```

Comment:
Exactly.  What happens when you decrypt forged or corrupted encryptions?  Or if you try to decrypt with the wrong secret key.